provider/core/src/resource.rs

```diff
@@ -114,9 +114,10 @@ impl ResourceKey {
     }
 
     #[doc(hidden)]
-    pub const fn try_new(path: &'static str) -> Option<Self> {
+    // Error is a str of the expected character class and the index where it wasn't encountered
+    pub const fn construct_internal(path: &'static str) -> Result<Self, (&'static str, usize)> {
```

Comment:
thought: we probably should use an enum for the error types? I guess a string is fine

Reply:
Well this is not really public, so I don't want to add another hidden API. This should get DCE'd so we don't have the strings in the binary.